src/components/LHNOptionsList/OptionRowLHNData.js

```diff
@@ -151,40 +151,50 @@ const personalDetailsSelector = (personalDetails) =>
-        withOnyx({
-            fullReport: {
-                key: (props) => ONYXKEYS.COLLECTION.REPORT + props.reportID,
-            },
-            reportActions: {
-                key: ({reportID}) => `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`,
-                canEvict: false,
-            },
-            personalDetails: {
-                key: ONYXKEYS.PERSONAL_DETAILS_LIST,
-                selector: personalDetailsSelector,
-            },
-            preferredLocale: {
-                key: ONYXKEYS.NVP_PREFERRED_LOCALE,
-            },
-            policies: {
-                key: ONYXKEYS.COLLECTION.POLICY,
-            },
-        }),
-        withOnyx({
-            parentReportActions: {
-                key: ({fullReport}) => `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${fullReport.parentReportID}`,
-                canEvict: false,
-            },
-        }),
-    )(OptionRowLHNData),
+export default compose(
+    withCurrentReportID,
+    withReportCommentDrafts({
+        propName: 'comment',
+        transformValue: (drafts, props) => {
+            const draftKey = `${ONYXKEYS.COLLECTION.REPORT_DRAFT_COMMENT}${props.reportID}`;
+            return lodashGet(drafts, draftKey, '');
+        },
+    }),
+    withOnyx({
+        fullReport: {
+            key: (props) => ONYXKEYS.COLLECTION.REPORT + props.reportID,
+        },
+        reportActions: {
+            key: ({reportID}) => `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`,
+            canEvict: false,
+        },
+        personalDetails: {
+            key: ONYXKEYS.PERSONAL_DETAILS_LIST,
+            selector: personalDetailsSelector,
+        },
+        preferredLocale: {
+            key: ONYXKEYS.NVP_PREFERRED_LOCALE,
+        },
+        policies: {
+            key: ONYXKEYS.COLLECTION.POLICY,
+        },
+    }),
+    /** The second withOnyx is needed here since fullReport only available after we successfully retrieve it from the previous withOnyx */
+    withOnyx({
+        parentReportActions: {
+            key: ({fullReport}) => `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${fullReport.parentReportID}`,
+            canEvict: false,
+        },
+    }),
+)(
+    memo(OptionRowLHNData, (prevProps, nextProps) => {
+        const prevParentReportActions = prevProps.parentReportActions[prevProps.fullReport.parentReportActionID];
+        const nextParentReportActions = nextProps.parentReportActions[nextProps.fullReport.parentReportActionID];
+        const prevPolicy = lodashGet(prevProps.policies, [`${ONYXKEYS.COLLECTION.POLICY}${prevProps.fullReport.policyID}`], '');
+        const nextPolicy = lodashGet(nextProps.policies, [`${ONYXKEYS.COLLECTION.POLICY}${nextProps.fullReport.policyID}`], '');
+        if (prevParentReportActions !== nextParentReportActions || prevPolicy !== nextPolicy) {
+            return false;
+        }
+        return _.omit(prevProps, 'parentReportActions', 'policies') === _.omit(nextProps, 'parentReportActions', 'policies');
```

Comment:
We should be using `Object.is` and not `===`. Object.is should be applied to each prop. It may be easier to just use `_.isEqual`.